### PR TITLE
Feature: suport public ip auto detected

### PR DESCRIPTION
**Pull request by njucjc**

### CODE REVIEW

File: pkg/ravencontroller/controllers/gateway_controller.go [hunk 1]
```diff
@@ -155,7 +155,11 @@ func (r *GatewayReconciler) electActiveEndpoint(nodeList corev1.NodeList, gw *ra
 
 	// the current active endpoint is still competent.
 	if checkActive(gw.Status.ActiveEndpoint) {
-		return gw.Status.ActiveEndpoint.DeepCopy()
+		for _, v := range gw.Spec.Endpoints {
```
Comment (DrmagicE) [inline pkg/ravencontroller/controllers/gateway_controller.go:158]: I think `checkActive` has already assured that `gw.Status.ActiveEndpoint` is in the `gw.Spec.Endpoints` list.  Did you find any bugs with the original code?

**Comment by DrmagicE** [comment]
Please merge two commits into one.

File: pkg/ravencontroller/controllers/gateway_controller.go (hunk 1 above)
Comment (njucjc) [reply pkg/ravencontroller/controllers/gateway_controller.go:158]: After the **controller manager** elects the `ActiveEndpoint`, **Raven** will update the `publicIP` of the `Active Endpoint`，so we should use the copy of updated one.

**Comment by njucjc** [comment]
I open a new  [pull request](https://github.com/openyurtio/raven-controller-manager/pull/5),  this one will be closed

File: pkg/ravencontroller/controllers/gateway_controller.go (hunk 1 above)
Comment (DrmagicE) [reply pkg/ravencontroller/controllers/gateway_controller.go:158]: Got it.

**Pull request closed without merging**